fiftyone/server/query.py

```diff
@@ -607,11 +605,11 @@ async def serialize_dataset(
     update_last_loaded_at=False,
 ) -> Dataset:
     def run():
-        if not fod.dataset_exists(dataset_name):
+        try:
+            dataset = fo.load_dataset(dataset_name, reload=True)
+        except fod.DatasetNotFoundError:
             return None
 
-        dataset = fo.Dataset(dataset_name, _create=False, _force_load=True)
-        dataset.reload()
```

Comment:
@benjaminpkane inspect this to see if the original bug this code fixed would still show up or not (either singleton_cache mode or not)